### ISSUE TITLE
Pre-processing data

### ISSUE THREAD

**Issue by dkomer** [wang-bin/QtAV#421]
Just getting my feet wet and looking at QtAV for an upcoming project.. let's assume we wanted to perform some basic in-memory decryption/atom-changing/etc. for a mp4 file on disc or sent over network stream, where would be the most logical place to insert that byte-transformation code before QtAV actually does anything with it, but in a way that it will still work perfectly for seeking etc.?

Imagine like a simple xor cipher, just need to operate on it before QtAV tries to process it as legitimate video data.

In other words- where is the data actually read and return the bytes (so that we can modify it in-memory before bytes are returned)? :)

Note, seems like a ffmpeg-related term for this is "custom io context": http://miphol.com/muse/2014/03/custom-io-with-ffmpeg.html

Not sure if this is exactly an "issue", more like a question for advice... but this seems to be the most active discussion group at the moment and I'm sure this relates to legitimate use-cases/issues I'm not aware of :)

**Comment by dkomer** [comment]
Oh... it seems this is as simple as creating a custom AVInput and process the data in read() before writing that to *data. Is there a basic example of this somewhere? Would be helpful to make sure I'm on the right way :)

**Comment by wang-bin** [comment]
You are right. QIODeviceInput.cpp

**Comment by dkomer** [comment]
Thanks! I'm looking for where to pass the AVInput (instead of filename) to AVPlayer ... but I don't see an example for that. Just by naming convention it seems that I'd need to call AVPlayer::setInput(), but that doesn't seem to be called anywhere? Any help is appreciated! Forgive me, I'm coming from C and no experience with C++... trying to catch up in tracing the Qt and C++ code :)

BTW- would it be better to ask these things on the google group?

**Comment by wang-bin** [comment]
`AVPlayer::setInput()` should work.

I have to use vpn to visit google sites in China, so github is preferred.
